### PR TITLE
eslint-config: add react configuration

### PR DESCRIPTION
**Pull request by sigorilla**
@ikokostya @twoRoger

### CODE REVIEW

File: packages/eslint-config-loris/README.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ npm install --save-dev eslint eslint-config-loris
 
 ## Usage
 
-Include config in your [.eslintrc.js](http://eslint.org/docs/user-guide/configuring#configuration-file-formats) file:
+Include config in your [.eslintrc.js](https://eslint.org/docs/user-guide/configuring#configuration-file-formats) file:
```
Comment (ikokostya) [inline packages/eslint-config-loris/README.md]: `eslint.org` is not available by https.

File: packages/eslint-config-loris/react.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        'react/no-is-mounted': 'error',
+        'react/no-multi-comp': 'warn',
+        'react/no-render-return-value': 'error',
+        'react/no-set-state': 'off',
+        'react/no-string-refs': 'error',
+        'react/no-unescaped-entities': 'error',
+        'react/no-unknown-property': 'error',
+        'react/no-unused-prop-types': 'error',
+        'react/no-will-update-set-state': 'error',
+        'react/prefer-es6-class': 'off',
```
Comment (ikokostya) [inline packages/eslint-config-loris/react.js]: Why don't use classes?

Comment (sigorilla) [inline packages/eslint-config-loris/react.js]: Yes, It's legacy. I'll fix this

File: packages/eslint-config-loris/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
   ],
   "peerDependencies": {
     "eslint": "^3.2.0",
-    "eslint-plugin-react": "^7.0.0"
+    "eslint-plugin-react": "^7.1.0"
```
Comment (ikokostya) [inline packages/eslint-config-loris/package.json]: Can you revert this changes? We up version in separate commit with changelog modification, e.g. https://github.com/ymaps/codestyle/commit/67b04a410208761c13372237864a89cfaf76f37b

**Comment by ikokostya** [comment]
@lemmy- Could you take a look?

File: packages/eslint-config-loris/package.json (hunk 3 above)
Comment (sigorilla) [inline packages/eslint-config-loris/package.json]: It's not version of our package. It's version of `eslint-plugin-react`

File: packages/eslint-config-loris/react.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        'react/forbid-foreign-prop-types': 'off',
+        'react/no-array-index-key': 'off',
+        'react/no-children-prop': 'off',
+        'react/no-danger': 'off',
+        'react/no-danger-with-children': 'error',
+        'react/no-deprecated': 'error',
+        'react/no-did-mount-set-state': 'error',
+        'react/no-did-update-set-state': 'error',
+        'react/no-direct-mutation-state': 'error',
+        'react/no-find-dom-node': 'off',
```
Comment (lemmy-) [inline packages/eslint-config-loris/react.js]: Do you use this a lot?

File: packages/eslint-config-loris/react.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        }
+    },
+
+    rules: {
+        // http://eslint.org/docs/rules/#stylistic-issues
+        'jsx-quotes': ['error', 'prefer-double'],
+
+        // https://github.com/yannickcr/eslint-plugin-react#list-of-supported-rules
+        'react/default-props-match-prop-types': 'off',
+        'react/display-name': 'error',
```
Comment (lemmy-) [inline packages/eslint-config-loris/react.js]: I think this rule can be up to the project team.

File: packages/eslint-config-loris/react.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        'react/no-redundant-should-component-update': 'off',
+        'react/no-render-return-value': 'error',
+        'react/no-set-state': 'off',
+        'react/no-string-refs': 'error',
+        'react/no-unescaped-entities': 'error',
+        'react/no-unknown-property': 'error',
+        'react/no-unused-prop-types': 'error',
+        'react/no-will-update-set-state': 'error',
+        'react/prefer-es6-class': ['error', 'always'],
+        'react/prefer-stateless-function': 'off',
```
Comment (lemmy-) [inline packages/eslint-config-loris/react.js]: This rule is useful.

File: packages/eslint-config-loris/react.js (hunk 4 above)
Comment (sigorilla) [inline packages/eslint-config-loris/react.js]: Yes

File: packages/eslint-config-loris/react.js (hunk 5 above)
Comment (sigorilla) [inline packages/eslint-config-loris/react.js]: As all another rules... I think for development we should write display name, because it helps to find components in devtools.

File: packages/eslint-config-loris/react.js (hunk 6 above)
Comment (sigorilla) [inline packages/eslint-config-loris/react.js]: Yeap. I'll add it with `{ignorePureComponents: true}`. It's ok?

File: packages/eslint-config-loris/package.json (hunk 3 above)
Comment (ikokostya) [inline packages/eslint-config-loris/package.json]: Ah, sorry.

File: packages/eslint-config-loris/react.js (hunk 5 above)
Comment (lemmy-) [inline packages/eslint-config-loris/react.js]: In the end this rule affects nothing but your convenience of debugging. It doesn't pinpoint any bad practices, potential errors or code style issues. Though it is useful to have this property, not every component may need it (like very simple ones).

Comment (sigorilla) [inline packages/eslint-config-loris/react.js]: Ok, I understand your point and agree with you. So this rule will be turned off.

File: packages/eslint-config-loris/react.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        'react/jsx-max-props-per-line': 'off',
+        'react/jsx-no-bind': 'off',
+        'react/jsx-no-comment-textnodes': 'error',
+        'react/jsx-no-duplicate-props': 'error',
+        'react/jsx-no-literals': 'off',
+        'react/jsx-no-target-blank': 'off',
+        'react/jsx-no-undef': 'error',
+        'react/jsx-pascal-case': 'error',
+        'react/jsx-sort-props': 'off',
+        'react/jsx-space-before-closing': ['error', 'always'],
```
Comment (sigorilla) [inline packages/eslint-config-loris/react.js]: Nice catch!

File: packages/eslint-config-loris/react.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        'react/no-children-prop': 'off',
+        'react/no-danger': 'off',
+        'react/no-danger-with-children': 'error',
+        'react/no-deprecated': 'error',
+        'react/no-did-mount-set-state': 'error',
+        'react/no-did-update-set-state': 'error',
+        'react/no-direct-mutation-state': 'error',
+        'react/no-find-dom-node': 'off',
+        'react/no-is-mounted': 'error',
+        'react/no-multi-comp': 'warn',
```
Comment (lemmy-) [inline packages/eslint-config-loris/react.js]: This is the only warning in the whole preset. I suggest that we use either 'off' or 'error'.

**Comment by lemmy-** [comment]
Everything else LGTM :)

**Comment by ikokostya** [comment]
So, now all projects (even which not used `React`) must install `eslint-plugin-react` package. Maybe create separate package for it? 
/cc @hevil @dodev @sigorilla

**Comment by ikokostya** [comment]
For example `eslint-config-loris-react`. Then in config file:

```js
module.exports = {
    extends: ['loris/es6', 'loris-react'],
    root: true
};
```

**Comment by sigorilla** [comment]
Maybe we can move `eslint-plugin-react` to [`optionalDependencies`](https://docs.npmjs.com/files/package.json#optionaldependencies)?

**Comment by ikokostya** [comment]
If package from `optionalDependencies` can't be installed, all installation will be complete. But in out case it should fail.

**Comment by ikokostya** [comment]
Moreover in most cases `eslint-plugin-react` from `optionalDependencies` will be installed.

**Comment by sigorilla** [comment]
Ok, I agree to create another package
